adr/0026-schema.md

```diff
@@ -0,0 +1,102 @@
+
+At create time Zarf will package both a `zarf.yaml` and a `zarfv1.yaml`. If a `zarfv1.yaml` exists Zarf will use that. If a `zarfv1.yaml` does not exist, then Zarf will know that the package was created prior to v1 and use the regular `zarf.yaml`. If the package is deployed with v0 it will read the `zarf.yaml` as normal even if the package has a `zarfv1.yaml`. This will make it simpler to drop deprecated items with migration paths from the v1 schema while remaining backwards compatible as those deprecated items will exist in the `zarf.yaml`
+
+When Zarf introduces new keys that they are not ready to promise long term support for they will mark them as experimental in the schema. A key is assumed to be stable if it's not experimental or deprecated.
+
+There are several other keys Zarf will deprecate which will have automated migrations to new fields
+- `.metadata.aggregateChecksum` -> `.build.aggregateChecksum`
+- Metadata fields `image`, `source`, `documentation`, `url`, `authors`, `vendors` -> will become a map of `annotations`
+- `noWait` -> `wait` which will default to true. This change will happen on both `.components.manifests` and `components.charts`
+- `yolo` -> `airgap` which will default to true
+- `.components.actions.[default/onAny].maxRetries` -> `.components.actions.[default/onAny].retries`
+- charts will change to avoid [current confusion with keys](https://github.com/defenseunicorns/zarf/issues/2245). Exactly one of the following field will exist for each `components.charts`.
+```yaml
+  helm:
+    url: https://stefanprodan.github.io/podinfo
+    name: podinfo # replaces repoName since it's only applicable for helm repos
+
+  git:
+    url: https://stefanprodan.github.io/podinfo
+    path: charts/podinfo
+
+  oci:
+    url: oci://ghcr.io/stefanprodan/charts/podinfo
+
+  local:
+   path: chart
+```
+- `actions.wait` will be deprecated in favor of a wait list per component that is run on deploy. `action.wait` blocks that are `onDeploy.After` can be auto migrated and will work if deployed on v1. `wait` blocks that are not on `onDeploy.After` will not be auto migrated and the package will error out if deployed on v1. Zarf v1 will use kstatus for wait functionality instead of `kubectl wait`. `apiVersion` will be added as a required field on `wait`, and the `apiVersion` will be automatically set to the most recent `apiVersion` if the waited for resource is not a custom resource. Users can still wait on custom resources, but the resources must implement kStatus. `wait.condition` will be removed from the schema and Zarf will assume it is waiting for `ready`. A `wait` block will not be auto migrated if `condition` is not `set` to  `ready`, `available` or equivalent.
```

Comment:
1. Existence of resources should be [possible through kstatus](https://github.com/kubernetes-sigs/cli-utils/blob/master/pkg/kstatus/README.md#statuses), but we will need to make sure we allow for more than just current. Could you comment on #2718 asking about this use case. 
3. I have not tested it but I assume [kstatus](https://github.com/kubernetes-sigs/cli-utils/blob/master/pkg/kstatus/README.md#kstatus) has a current status for stateful sets which will cover that situation
4. Currently undecided from the team, but yes it should be explicitly stated in the ADR. Could you give us situations you see `http` waits used that would be hard to replace in a Kubernetes native way? One worry the team has about http waits is that we're encouraging users to use Zarf features rather than k8s features, expanding the scope for the tool

Reply:
FYI there is a kstatus implementation for all Kubernetes native resources, the only question is if CRDs implement the necessary fields for kstatus